pkg/nfd-worker/nfd-worker.go

```diff
@@ -761,7 +816,7 @@ func (m *nfdWorker) updateNodeFeatureObject(labels Labels) error {
 		nfrUpdated := nfr.DeepCopy()
 		nfrUpdated.Annotations = map[string]string{nfdv1alpha1.WorkerVersionAnnotation: version.Get()}
 		nfrUpdated.Labels = map[string]string{nfdv1alpha1.NodeFeatureObjNodeNameLabel: nodename}
-		nfrUpdated.OwnerReferences = ownerRefs
+		nfrUpdated.OwnerReferences = m.ownerReference
```

Comment:
@ArangoGutierrez question regarding L825. if no change to NF spec will it still update the object to set owner reference ? just want to make sure that if we update NFD in the cluster (e.g helm upgrade -n <nfd-namespace> ....), owner reference will be added to existing NF CRs.

Reply:
`DeepEqual` evaluates the entire object, so the answer is yes, if this is a new pod, because XYZ reason, we would update the NF with the new ownerRef (if this changed)

Reply:
Yes, it updates the object if anything changes (including the object metadata)